### ISSUE TITLE
Upgrading from 2018.10 to 2019.3+ fails when package configuration contains forward slashes

### ISSUE THREAD

**Issue by flin-8** [OctopusDeploy/Issues#5716]
# The bug

When moving package folders during the upgrade, the code that determines whether a folder is a sub-folder of another doesn't work as expected when the package configuration contain forward slashes ("/"). As a result the upgrade fails as Octopus tries to move a folder into its child.

The error will look something like:
```
Moving the built in package feed failed. Unable to access a directory while moving: Access to the path 'C:/Octopus/Packages\Spaces-1' is denied.. You can still move packages manually from C:/Octopus/Packages into the new location for the built in feed: C:/Octopus/Packages\Spaces-1\feeds-builtin, and then use the re-index feature as required. There may also be packages in C:\Octopus\PackagesTemp
Error occured when executing scripts, transaction will be rolled back
Upgrade failed due to an unexpected exception:
System.IO.IOException: Access to the path 'C:/Octopus/Packages\Spaces-1' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Octopus.Core.UpgradeScripts.Script0180BuiltInFeedMigration.MovePackagesAndFiles(String sourceDirectory, String destinationDirectory)
   at Octopus.Core.UpgradeScripts.Script0180BuiltInFeedMigration.MoveSubdirectoriesAndFiles(String sourceDirectory, Maybe`1 tempPackagesDirectory, String destinationDirectory)
   at Octopus.Core.UpgradeScripts.Script0180BuiltInFeedMigration.Execute(IDbCommandExecutor commandExecutor, IUpgradeLog log)
   at Octopus.Core.Initialization.DbCommandUpgradeScriptResolver.Execute(Func`1 commandFactory, IUpgradeLog log)
   at Octopus.Core.Initialization.DatabaseUpgrader.OctopusUpgradeScriptExecutor.<>c__DisplayClass6_0.<ExecuteDbCommandUpgradeScript>b__0(Func`1 commandFactory)
   at DbUp.Engine.Transactions.SingleTrasactionStrategy.Execute(Action`1 action)
   at Octopus.Core.Initialization.DatabaseUpgrader.OctopusUpgradeScriptExecutor.ExecuteDbCommandUpgradeScript(IConnectionManager connectionManager, IJournal journal, IUpgradeLog log, IDbCommandUpgradeScript upgradeScript)
   at Octopus.Core.Initialization.DatabaseUpgrader.OctopusUpgradeScriptExecutor.Execute(SqlScript script, IDictionary`2 variables)
   at DbUp.Engine.UpgradeEngine.PerformUpgrade()
```
## Steps to reproduce

1. On 2018.10, Go to **Configuration** -> **Settings** -> **Server Folders**
2. Under **Packages Directory**, replace some "\\"s with "/"
3. Upgrade to 2019.3+

## Affected versions

**Octopus Server: 2019.3-2019.7** 

## Workarounds

Change "/" to "\\"s

## Links

https://secure.helpscout.net/conversation/884476226/46243?folderId=557082

**Comment by flin-8** [comment]
Release Note: Fix for Upgrading from 2018.10 to 2019.3+ fails when package configuration contains forward slashes

**Comment by flin-8** [comment]
PR: https://github.com/OctopusDeploy/OctopusDeploy/pull/4075
